distributed/shuffle/_rechunk.py

```diff
@@ -431,52 +432,108 @@ def _construct_graph(self) -> _T_LowLevelGraph:
         return dsk
 
 
+def _prechunk_for_partials(
```

Comment:
The function name implies to me that we apply the rechunking here, could you rename to something that tells us that we are only calculating the chunks?

Reply:
Done